### PR TITLE
Minor Ltac2 documentation fix: type parameters are optional.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation fix

### CODE REVIEW

File: doc/sphinx/proof-engine/ltac2.rst
Comment (Zimmi48) [inline doc/sphinx/proof-engine/ltac2.rst]: @ppedrot This is still incorrect: the command expects an unqualified identifier. However, this looks more like a Ltac2 bug than a documentation bug to me. Indeed, how is one supposed to expand, e.g., the `exn` type if it can only be an unqualified identifier?

```coq
From Ltac2 Require Import Ltac2.

Ltac2 Type exn := [ Foo ].

Ltac2 Eval Control.zero Foo.
```

gives the following error message:

```
This expression has type exn but an expression was expected of type Init.exn
```

**Review by cpitclaudel — APPROVED** [review]
LGTM, but I think I should wait for @ppedrot to respond before I merge, right?

**Comment by Zimmi48** [comment]
Yes.

File: doc/sphinx/proof-engine/ltac2.rst
Comment (ppedrot) [reply doc/sphinx/proof-engine/ltac2.rst]: The syntax used for extension is not the same, it needs to use `::=` instead of `:=`. What you have done here is defining a new type with a brand new constructor, hence the error.

**Review by ppedrot — APPROVED** [review]

File: doc/sphinx/proof-engine/ltac2.rst
Comment (Zimmi48) [reply doc/sphinx/proof-engine/ltac2.rst]: Then the doc was incorrect since it was showing a syntax with only one colon. Fixed now.

**Comment by Zimmi48** [comment]
Now ready.

**Comment by Zimmi48** [comment]
@cpitclaudel or @ppedrot Ready to merge.

**Pull request merged**
